api/api.js

```diff
@@ -0,0 +1,91 @@
+const storage = require('../storage/storage.js')
+const Security = require('./security');
+const cryptoCurrencies = require('./cryproCurrencies')
+
+
+const emailAlreadyExist = email => {
+    let users_data = storage.get()
+    return users_data.length != 0 && users_data.every(user => user.email == email)
+}
+
+const isCorrectEmail = email => email.match(/^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/)
+
+const api = {
+    '/user/create/POST': async ({email, password}) => {
+        if(!(email && password)) {
+            return {
+                body: `Error, your E-mail or password field is empty`,
+                status: 400
+            }
+        }
+        if(!isCorrectEmail(email)) {
+            return {
+                body: `Error, incorrect email`,
+                status: 400
+            }
+        }
+        if(emailAlreadyExist(email)){
+            return {
+                body: `Error, user with this E-mail adress already exists`,
+                status: 409
+            }
+        }
+        const hashed_password = await Security.hashPassword(password)
+        const token = await Security.generateToken()
+        await storage.insert({
+            email: email,
+            password: hashed_password,
+            token: token
+        })
+        return {
+            body: `Created user with E-mail: ${email}`,
+            status: 201
+        }
+    },
+
+    '/user/login/POST': async ({email, password}) => {
+        if(!(email && password)) {
+            return {
+                body: `Error, your E-mail or password field is empty`,
+                status: 400
+            }
+        }
+        if(!emailAlreadyExist(email)){
+            return {
+                body: `Error, user with this email does not exist`,
+                status: 400
+            }
+        }
+        const user = storage.get(email)
+        if(!await Security.verifyPassword(password, user.password)){
+            return {
+                body: `Error, password is invalid`,
+                status: 400
+            }
+        }
+        return {
+                body: user.token,
+                status: 200
+            }
+    },
+
+    '/btcRate/GET': async (_, {Authorization}) => {
```

Comment:
Is the `Authorization` a field in the request body? As I know the usual JWT approach is to pass the JWT in the `Authorization` request header. The standard way with the passport.js for example: https://www.freecodecamp.org/news/learn-how-to-handle-authentication-with-node-using-passport-js-4a56ed18e81e/

Reply:
I just didn't use JWT auth method, and made my own. In our task there wasn't any requirements for JWT